plugins/discovery-ec2/src/main/java/org/elasticsearch/cloud/aws/Ec2Module.java

```diff
@@ -27,4 +30,18 @@
     protected void configure() {
         bind(AwsEc2Service.class).to(AwsEc2ServiceImpl.class).asEagerSingleton();
     }
+
+    /**
+     * Check if discovery is meant to start
+     * @return true if we can start discovery features
+     */
+    public static boolean isDiscoveryReady(Settings settings, ESLogger logger) {
```

Comment:
can we rename this to `isEc2DiscoveryActive`? ready feels weird given the check. Also, why is this on the module? the plugin uses it , why not just put it there?